ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -74,7 +75,7 @@ public MessageInfo(StoreKey key, long size, boolean deleted, boolean ttlUpdated,
    */
   public MessageInfo(StoreKey key, long size, boolean deleted, boolean ttlUpdated, long expirationTimeInMs,
       short accountId, short containerId, long operationTimeMs) {
-    this(key, size, deleted, ttlUpdated, expirationTimeInMs, null, accountId, containerId, operationTimeMs);
+    this(key, size, deleted, ttlUpdated, expirationTimeInMs, null, accountId, containerId, operationTimeMs, (short) -1);
```

Comment:
What is the though to use -1 as default value? 
I feel it's more natural to use 0 as default value so that existing Put/Delete/TtlUpdate all get updateVersion 0, which is expected. 
With default 0, no need to introduce `hasUpdateVersion`. 
0 makes everything auto backwards compatible.

Reply:
I was still skeptical of your proposed numbering scheme at the time, but after reviewing your "back-to-life" versioning scheme (incrementing every-time a blob is undeleted, so that TTL-UPDATEs and DELETEs of a blob inherit the latest updateVersion) I think this makes sense

Reply:
Cool. And it would be great if add some comments about this back-to-life" versioning scheme.